dwibiascorrect: error when using the `-debug` flag

**The bug**

Running `dwibiascorrect` with option flag `-debug` raises error.

**To Reproduce**

Command:
```sh
/home/localadmin/Softwares/miniconda3/bin/dwibiascorrect ants -mask /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz -bias diffusion_denoised_biasfield.mif -debug -force /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_biascorrect/diffusion_denoised_biascorr.mif
```

Terminal output:
```output
dwibiascorrect: 
dwibiascorrect: Note that this script makes use of commands / algorithms that have relevant articles for citation; INCLUDING FROM EXTERNAL SOFTWARE PACKAGES. Please consult the help page (-help option) for more information.
dwibiascorrect: 
dwibiascorrect: [DEBUG] path.script_subdir_name() (from algorithm.py:69): dwibiascorrect
dwibiascorrect: [WARNING] Output file 'diffusion_denoised_biasfield.mif' already exists; will be overwritten at script completion
dwibiascorrect: Generated scratch directory: /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/
dwibiascorrect: [DEBUG] path.from_user() (from dwibiascorrect:50): /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif -> /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif
dwibiascorrect: [DEBUG] path.to_scratch() (from dwibiascorrect:50): in.mif -> /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/in.mif
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): mrconvert -> mrconvert
dwibiascorrect: [DEBUG] run.version_match() (from run.py:330): Version-matched executable for mrconvert: /home/localadmin/Softwares/miniconda3/bin/mrconvert
dwibiascorrect: [DEBUG] run._shebang() (from run.py:340): File "/home/localadmin/Softwares/miniconda3/bin/mrconvert": Not a text file
dwibiascorrect: [DEBUG] run.command() (from dwibiascorrect:50): To execute: [['/home/localadmin/Softwares/miniconda3/bin/mrconvert', '/home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif', '/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/in.mif', '-info']]
Command:  mrconvert /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/in.mif
          mrconvert: [INFO] opening image "/home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif"...
          mrconvert: [INFO] image "/home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif" opened with dimensions 96x96x38x129, voxel spacing 2.2083300000000001x2.2083300000000001x3x5.9000000000000004, datatype Float32LE
          mrconvert: [INFO] found 129x4 diffusion gradient table
          mrconvert: [INFO] creating image "/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/in.mif"...
          mrconvert: [INFO] image "/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/in.mif" created with dimensions 96x96x38x129, voxel spacing 2.2083300000000001x2.2083300000000001x3x5.9000000000000004, datatype Float32LE
          mrconvert: [100%] copying from "/home/loca...oise/diffusion_denoised.mif" to "/home/loca...ascorrect-tmp-XEGOZD/in.mif"
dwibiascorrect: [DEBUG] path.from_user() (from dwibiascorrect:52): /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz -> /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz
dwibiascorrect: [DEBUG] path.to_scratch() (from dwibiascorrect:52): mask.mif -> /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mask.mif
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): mrconvert -> mrconvert
dwibiascorrect: [DEBUG] run.version_match() (from run.py:330): Version-matched executable for mrconvert: /home/localadmin/Softwares/miniconda3/bin/mrconvert
dwibiascorrect: [DEBUG] run._shebang() (from run.py:340): File "/home/localadmin/Softwares/miniconda3/bin/mrconvert": Not a text file
dwibiascorrect: [DEBUG] run.command() (from dwibiascorrect:52): To execute: [['/home/localadmin/Softwares/miniconda3/bin/mrconvert', '/home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz', '/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mask.mif', '-datatype', 'bit', '-info']]
Command:  mrconvert /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mask.mif -datatype bit
          mrconvert: [INFO] opening image "/home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz"...
          mrconvert: [INFO] image "/home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz" opened with dimensions 96x96x38, voxel spacing 2.2083332538604736x2.2083332538604736x3, datatype Float32LE
          mrconvert: [INFO] no valid diffusion gradient table found
          mrconvert: [INFO] error importing diffusion gradient table for image "/home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz"
          mrconvert: [100%] uncompressing image "/home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz"
          mrconvert: [INFO] creating image "/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mask.mif"...
          mrconvert: [INFO] image "/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mask.mif" created with dimensions 96x96x38, voxel spacing 2.2083332538604736x2.2083332538604736x3, datatype Bit
          mrconvert: [100%] copying from "/home/loca...imask/dwi_brain_mask.nii.gz" to "/home/loca...correct-tmp-XEGOZD/mask.mif"
dwibiascorrect: Changing to scratch directory (/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/)
dwibiascorrect: [DEBUG] path.name_temporary() (from image.py:33): /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mrtrix-tmp-ti2iWL.json
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): mrinfo -> mrinfo
dwibiascorrect: [DEBUG] run.version_match() (from image.py:34): Version-matched executable for mrinfo: /home/localadmin/Softwares/miniconda3/bin/mrinfo
dwibiascorrect: [DEBUG] run.exe_name() (from image.py:34): /home/localadmin/Softwares/miniconda3/bin/mrinfo -> /home/localadmin/Softwares/miniconda3/bin/mrinfo
dwibiascorrect: Loading header for image file 'in.mif'
dwibiascorrect: [DEBUG] image.__init__() (from dwibiascorrect:59): ['/home/localadmin/Softwares/miniconda3/bin/mrinfo', 'in.mif', '-json_all', '/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mrtrix-tmp-ti2iWL.json']
dwibiascorrect: [DEBUG] image.__init__() (from dwibiascorrect:59): {'_name': 'in.mif', '_size': [96, 96, 38, 129], '_spacing': [2.20833, 2.20833, 3.0, 5.9], '_strides': [1, 2, 3, 4], '_format': 'MRtrix', '_datatype': 'Float32LE', '_intensity_offset': 0.0, '_intensity_scale': 1.0, '_transform': [[0.999999943880058, 5.33884694434427e-08, 0.000335022184972424, -102.900541691394], [1.57276657824972e-05, 0.998889975305693, -0.0471043187585057, -93.3131521331962], [-0.000334652837870155, 0.0471043228779298, 0.998889919317656, -23.3364965748897], [0.0, 0.0, 0.0, 1.0]], '_keyval': {'command_history': ["mrconvert -stride '1,2,3,4' -force -fslgrad /output_dir/cmp/sub-01/ses-01/dwi/sub-01_ses-01_desc-cmp_dwi.bvec /output_dir/cmp/sub-01/ses-01/dwi/sub-01_ses-01_desc-cmp_dwi.bval -quiet /output_dir/cmp/sub-01/ses-01/dwi/sub-01_ses-01_desc-cmp_dwi.nii.gz diffusion.mif  (version=3.0.2)", 'dwidenoise -mask /output_dir/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz -noise diffusion_noisemap.mif -force -debug /output_dir/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/mr_convert/diffusion.mif diffusion_denoised.mif  (version=3.0.2)', '/home/localadmin/Softwares/miniconda3/bin/mrconvert /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/in.mif -info  (version=3.0.2)'], 'comments': 'TE=1e+02;Time=160316.967;phase=1', 'dw_scheme': [[0.0, 0.0, 0.0, 0.0], [4.604794858e-11, -0.9999980001, -0.001999937115, 500.0], [1.455731894e-10, 2.590861083e-08, -1.0, 500.0], [-0.999998, -9.949299793e-11, -0.001999989269, 500.0], [0.7071057544, -0.7071063938, -0.00141422849, 1000.0], [0.7067542777, -5.561829258e-08, -0.707459109, 1000.0], [2.6698783e-10, -0.7067539959, -0.7074593906, 1000.0], [1.98246735e-10, 0.7067540551, -0.7074593314, 1000.0], [-0.7071061444, -0.7071060037, -0.001414231861, 1000.0], [-0.7067538111, 4.064903066e-08, -0.7074595752, 1000.0], [0.5770938972, -0.5770939732, -0.577862596, 1500.0], [0.5770936909, 0.5770944823, -0.5778622937, 1500.0], [-0.5770942262, -0.5770936115, -0.5778626287, 1500.0], [-0.5770938222, 0.5770937722, -0.5778628717, 1500.0], [4.701760879e-11, -0.999999875, -0.0004999499268, 2000.0], [1.455731894e-10, 2.590861083e-08, -1.0, 2000.0], [-0.999999875, -6.189255252e-11, -0.0004999995985, 2000.0], [0.8944271678, -0.44721332, -0.0005366216217, 2500.0], [0.8942481999, 1.233911471e-09, -0.4475713987, 2500.0], [0.4472140294, -0.894426813, -0.0005366778619, 2500.0], [0.447141854, -6.331739179e-08, -0.8944630581, 2500.0], [6.047983974e-10, -0.8942484278, -0.4475709434, 2500.0], [-1.547638207e-10, -0.4471420937, -0.8944629383, 2500.0], [6.038469206e-11, 0.4471424734, -0.8944627485, 2500.0], [-2.987575823e-10, 0.8942483962, -0.4475710065, 2500.0], [-0.4472142273, -0.8944267141, -0.0005366223904, 2500.0], [-0.4471422325, 2.979426711e-09, -0.894462869, 2500.0], [-0.8944272642, -0.4472131271, -0.0005366111842, 2500.0], [-0.8942481864, -1.203342597e-08, -0.4475714257, 2500.0], [0.8163150495, -0.4082259566, -0.4086334645, 3000.0], [0.8163149864, 0.4082255761, -0.4086339707, 3000.0], [0.4082262058, -0.816314944, -0.4086334263, 3000.0], [0.4081581457, -0.408157277, -0.816587145, 3000.0], [0.408157348, 0.4081578487, -0.8165872579, 3000.0], [0.4082253801, 0.8163154, -0.4086333403, 3000.0], [-0.4082260024, -0.8163148249, -0.4086338675, 3000.0], [-0.4081577425, -0.408157098, -0.816587436, 3000.0], [-0.4081578271, 0.4081579636, -0.816586961, 3000.0], [-0.4082254286, 0.8163152063, -0.4086336787, 3000.0], [-0.8163152896, -0.4082256582, -0.408633283, 3000.0], [-0.8163150614, 0.4082255313, -0.4086338657, 3000.0], [0.7071071135, -0.7071063605, -0.0003535496407, 4000.0], [0.7069299418, -5.319339341e-09, -0.7072835764, 4000.0], [-9.789377913e-11, -0.7069302433, -0.707283275, 4000.0], [1.567798364e-10, 0.7069298265, -0.7072836916, 4000.0], [-0.7071066491, -0.7071068248, -0.0003535998457, 4000.0], [-0.7069299747, 3.830753602e-08, -0.7072835434, 4000.0], [0.6665928365, -0.6665925637, -0.3336290519, 4500.0], [0.6664856674, -0.3333169916, -0.6668557852, 4500.0], [0.6664856527, 0.3333167604, -0.6668559155, 4500.0], [0.6665929437, 0.6665923207, -0.3336293233, 4500.0], [0.333317104, -0.6664855564, -0.66685584, 4500.0], [0.3333173941, 0.6664856233, -0.6668556281, 4500.0], [4.085725123e-11, -0.9999999012, -0.0004444170368, 4500.0], [1.455731894e-10, 2.590861083e-08, -1.0, 4500.0], [-0.3333166201, -0.6664858367, -0.6668558017, 4500.0], [-0.3333166057, 0.6664855235, -0.6668561219, 4500.0], [-0.6665924266, -0.666592734, -0.3336295307, 4500.0], [-0.6664858483, -0.3333171152, -0.6668555426, 4500.0], [-0.6664853425, 0.3333167888, -0.6668562113, 4500.0], [-0.6665924202, 0.666592617, -0.3336297773, 4500.0], [-0.9999999012, 3.742696584e-11, -0.0004444399636, 4500.0], [0.9486830278, -0.3162282667, -0.0004427363325, 5000.0], [0.9485630721, 2.321566786e-08, -0.3165882157, 5000.0], [0.3162277787, -0.9486831905, -0.0004427124981, 5000.0], [0.3162084303, -1.470391669e-08, -0.9486897431, 5000.0], [6.094271403e-11, -0.9485631972, -0.3165878406, 5000.0], [3.727802446e-10, -0.3160571874, -0.9487401406, 5000.0], [2.102303804e-10, 0.3162084941, -0.9486897218, 5000.0], [2.418275742e-10, 0.9485631902, -0.3165878617, 5000.0], [-0.3162281986, -0.9486830505, -0.0004427240341, 5000.0], [-0.3162086596, -2.213515469e-08, -0.9486896667, 5000.0], [-0.9486830937, -0.3162280692, -0.0004427716254, 5000.0], [-0.9485632352, -2.792397978e-09, -0.3165877268, 5000.0], [0.9044196305, -0.3014910643, -0.3018745932, 5500.0], [0.9044195348, 0.3014912943, -0.3018746504, 5500.0], [0.3014912367, -0.9044193748, -0.3018751873, 5500.0], [0.3015110819, -0.3013465199, -0.9045890462, 5500.0], [0.3014815607, 0.3014810386, -0.9045540625, 5500.0], [0.301491691, 0.9044192968, -0.3018749673, 5500.0], [-0.3014917901, -0.9044192099, -0.3018751286, 5500.0], [-0.3015110626, -0.3013465021, -0.9045890585, 5500.0], [-0.3014814747, 0.3014810781, -0.904554078, 5500.0], [-0.301491225, 0.904419444, -0.3018749917, 5500.0], [-0.9044195788, -0.3014910964, -0.301874716, 5500.0], [-0.9044196065, 0.3014912379, -0.3018744917, 5500.0], [0.5772539059, -0.577254112, -0.5775427416, 6000.0], [0.5772537687, 0.5772545673, -0.5775424235, 6000.0], [-0.5772541227, -0.5772538622, -0.5775427744, 6000.0], [-0.577254148, 0.5772543081, -0.5775423035, 6000.0], [0.8320501259, -0.5547003438, -0.0003413330173, 6500.0], [0.8319060448, 3.098869511e-08, -0.5549165096, 6500.0], [0.5546999144, -0.8320504122, -0.0003413904583, 6500.0], [0.554572481, -4.940954412e-08, -0.8321354236, 6500.0], [6.072960914e-10, -0.8319059468, -0.5549166565, 6500.0], [-3.666727552e-10, -0.5545724001, -0.8321354776, 6500.0], [2.11767819e-10, 0.5545717321, -0.8321359228, 6500.0], [-3.600334341e-10, 0.831906107, -0.5549164164, 6500.0], [-0.5546999789, -0.8320503692, -0.0003413141108, 6500.0], [-0.5545722601, 2.913643411e-08, -0.8321355709, 6500.0], [-0.832050099, -0.5547003842, -0.0003413181217, 6500.0], [-0.8319060534, 3.281216423e-10, -0.5549164967, 6500.0], [0.8017208664, -0.53446813, -0.2675583495, 7000.0], [0.8016418421, -0.2672396717, -0.5347460283, 7000.0], [0.8016418981, 0.2672394531, -0.5347460537, 7000.0], [0.8017207264, 0.534468159, -0.267558711, 7000.0], [0.5344681068, -0.8017208029, -0.2675585861, 7000.0], [0.53442475, -0.2671552934, -0.8018841786, 7000.0], [0.5343916032, 0.2672533946, -0.8018735795, 7000.0], [0.5344678393, 0.8017210021, -0.2675585238, 7000.0], [0.2672396447, -0.8016419395, -0.5347458958, 7000.0], [0.2672530582, -0.5343914545, -0.8018737907, 7000.0], [0.2672532447, 0.5343918496, -0.8018734652, 7000.0], [0.2672394535, 0.8016416949, -0.534746358, 7000.0], [-0.2672393738, -0.8016420235, -0.5347459053, 7000.0], [-0.2672529745, -0.5343914835, -0.8018737993, 7000.0], [-0.2672527998, 0.534391969, -0.801873534, 7000.0], [-0.2672395729, 0.8016417046, -0.5347462838, 7000.0], [-0.534468046, -0.8017208702, -0.2675585059, 7000.0], [-0.534424298, -0.2671553634, -0.8018844565, 7000.0], [-0.5343916999, 0.2672534444, -0.8018734985, 7000.0], [-0.5344680464, 0.801720919, -0.267558359, 7000.0], [-0.8017210357, -0.5344679886, -0.2675581246, 7000.0], [-0.8016417412, -0.2672397248, -0.5347461531, 7000.0], [-0.8016420535, 0.2672394791, -0.5347458078, 7000.0], [-0.8017211137, 0.5344675679, -0.2675587314, 7000.0], [2.651765635e-12, -0.9999999687, -0.0002500086681, 8000.0], [1.455731894e-10, 2.590861083e-08, -1.0, 8000.0], [-0.9999999687, -1.852948745e-11, -0.0002500002363, 8000.0]], 'mrtrix_version': '3.0.2'}}
dwibiascorrect: [DEBUG] path.name_temporary() (from image.py:33): /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mrtrix-tmp-njtCya.json
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): mrinfo -> mrinfo
dwibiascorrect: [DEBUG] run.version_match() (from image.py:34): Version-matched executable for mrinfo: /home/localadmin/Softwares/miniconda3/bin/mrinfo
dwibiascorrect: [DEBUG] run.exe_name() (from image.py:34): /home/localadmin/Softwares/miniconda3/bin/mrinfo -> /home/localadmin/Softwares/miniconda3/bin/mrinfo
dwibiascorrect: Loading header for image file 'in.mif'
dwibiascorrect: [DEBUG] image.__init__() (from image.py:175): ['/home/localadmin/Softwares/miniconda3/bin/mrinfo', 'in.mif', '-json_all', '/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mrtrix-tmp-njtCya.json']
dwibiascorrect: [DEBUG] image.__init__() (from image.py:175): {'_name': 'in.mif', '_size': [96, 96, 38, 129], '_spacing': [2.20833, 2.20833, 3.0, 5.9], '_strides': [1, 2, 3, 4], '_format': 'MRtrix', '_datatype': 'Float32LE', '_intensity_offset': 0.0, '_intensity_scale': 1.0, '_transform': [[0.999999943880058, 5.33884694434427e-08, 0.000335022184972424, -102.900541691394], [1.57276657824972e-05, 0.998889975305693, -0.0471043187585057, -93.3131521331962], [-0.000334652837870155, 0.0471043228779298, 0.998889919317656, -23.3364965748897], [0.0, 0.0, 0.0, 1.0]], '_keyval': {'command_history': ["mrconvert -stride '1,2,3,4' -force -fslgrad /output_dir/cmp/sub-01/ses-01/dwi/sub-01_ses-01_desc-cmp_dwi.bvec /output_dir/cmp/sub-01/ses-01/dwi/sub-01_ses-01_desc-cmp_dwi.bval -quiet /output_dir/cmp/sub-01/ses-01/dwi/sub-01_ses-01_desc-cmp_dwi.nii.gz diffusion.mif  (version=3.0.2)", 'dwidenoise -mask /output_dir/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz -noise diffusion_noisemap.mif -force -debug /output_dir/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/mr_convert/diffusion.mif diffusion_denoised.mif  (version=3.0.2)', '/home/localadmin/Softwares/miniconda3/bin/mrconvert /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/in.mif -info  (version=3.0.2)'], 'comments': 'TE=1e+02;Time=160316.967;phase=1', 'dw_scheme': [[0.0, 0.0, 0.0, 0.0], [4.604794858e-11, -0.9999980001, -0.001999937115, 500.0], [1.455731894e-10, 2.590861083e-08, -1.0, 500.0], [-0.999998, -9.949299793e-11, -0.001999989269, 500.0], [0.7071057544, -0.7071063938, -0.00141422849, 1000.0], [0.7067542777, -5.561829258e-08, -0.707459109, 1000.0], [2.6698783e-10, -0.7067539959, -0.7074593906, 1000.0], [1.98246735e-10, 0.7067540551, -0.7074593314, 1000.0], [-0.7071061444, -0.7071060037, -0.001414231861, 1000.0], [-0.7067538111, 4.064903066e-08, -0.7074595752, 1000.0], [0.5770938972, -0.5770939732, -0.577862596, 1500.0], [0.5770936909, 0.5770944823, -0.5778622937, 1500.0], [-0.5770942262, -0.5770936115, -0.5778626287, 1500.0], [-0.5770938222, 0.5770937722, -0.5778628717, 1500.0], [4.701760879e-11, -0.999999875, -0.0004999499268, 2000.0], [1.455731894e-10, 2.590861083e-08, -1.0, 2000.0], [-0.999999875, -6.189255252e-11, -0.0004999995985, 2000.0], [0.8944271678, -0.44721332, -0.0005366216217, 2500.0], [0.8942481999, 1.233911471e-09, -0.4475713987, 2500.0], [0.4472140294, -0.894426813, -0.0005366778619, 2500.0], [0.447141854, -6.331739179e-08, -0.8944630581, 2500.0], [6.047983974e-10, -0.8942484278, -0.4475709434, 2500.0], [-1.547638207e-10, -0.4471420937, -0.8944629383, 2500.0], [6.038469206e-11, 0.4471424734, -0.8944627485, 2500.0], [-2.987575823e-10, 0.8942483962, -0.4475710065, 2500.0], [-0.4472142273, -0.8944267141, -0.0005366223904, 2500.0], [-0.4471422325, 2.979426711e-09, -0.894462869, 2500.0], [-0.8944272642, -0.4472131271, -0.0005366111842, 2500.0], [-0.8942481864, -1.203342597e-08, -0.4475714257, 2500.0], [0.8163150495, -0.4082259566, -0.4086334645, 3000.0], [0.8163149864, 0.4082255761, -0.4086339707, 3000.0], [0.4082262058, -0.816314944, -0.4086334263, 3000.0], [0.4081581457, -0.408157277, -0.816587145, 3000.0], [0.408157348, 0.4081578487, -0.8165872579, 3000.0], [0.4082253801, 0.8163154, -0.4086333403, 3000.0], [-0.4082260024, -0.8163148249, -0.4086338675, 3000.0], [-0.4081577425, -0.408157098, -0.816587436, 3000.0], [-0.4081578271, 0.4081579636, -0.816586961, 3000.0], [-0.4082254286, 0.8163152063, -0.4086336787, 3000.0], [-0.8163152896, -0.4082256582, -0.408633283, 3000.0], [-0.8163150614, 0.4082255313, -0.4086338657, 3000.0], [0.7071071135, -0.7071063605, -0.0003535496407, 4000.0], [0.7069299418, -5.319339341e-09, -0.7072835764, 4000.0], [-9.789377913e-11, -0.7069302433, -0.707283275, 4000.0], [1.567798364e-10, 0.7069298265, -0.7072836916, 4000.0], [-0.7071066491, -0.7071068248, -0.0003535998457, 4000.0], [-0.7069299747, 3.830753602e-08, -0.7072835434, 4000.0], [0.6665928365, -0.6665925637, -0.3336290519, 4500.0], [0.6664856674, -0.3333169916, -0.6668557852, 4500.0], [0.6664856527, 0.3333167604, -0.6668559155, 4500.0], [0.6665929437, 0.6665923207, -0.3336293233, 4500.0], [0.333317104, -0.6664855564, -0.66685584, 4500.0], [0.3333173941, 0.6664856233, -0.6668556281, 4500.0], [4.085725123e-11, -0.9999999012, -0.0004444170368, 4500.0], [1.455731894e-10, 2.590861083e-08, -1.0, 4500.0], [-0.3333166201, -0.6664858367, -0.6668558017, 4500.0], [-0.3333166057, 0.6664855235, -0.6668561219, 4500.0], [-0.6665924266, -0.666592734, -0.3336295307, 4500.0], [-0.6664858483, -0.3333171152, -0.6668555426, 4500.0], [-0.6664853425, 0.3333167888, -0.6668562113, 4500.0], [-0.6665924202, 0.666592617, -0.3336297773, 4500.0], [-0.9999999012, 3.742696584e-11, -0.0004444399636, 4500.0], [0.9486830278, -0.3162282667, -0.0004427363325, 5000.0], [0.9485630721, 2.321566786e-08, -0.3165882157, 5000.0], [0.3162277787, -0.9486831905, -0.0004427124981, 5000.0], [0.3162084303, -1.470391669e-08, -0.9486897431, 5000.0], [6.094271403e-11, -0.9485631972, -0.3165878406, 5000.0], [3.727802446e-10, -0.3160571874, -0.9487401406, 5000.0], [2.102303804e-10, 0.3162084941, -0.9486897218, 5000.0], [2.418275742e-10, 0.9485631902, -0.3165878617, 5000.0], [-0.3162281986, -0.9486830505, -0.0004427240341, 5000.0], [-0.3162086596, -2.213515469e-08, -0.9486896667, 5000.0], [-0.9486830937, -0.3162280692, -0.0004427716254, 5000.0], [-0.9485632352, -2.792397978e-09, -0.3165877268, 5000.0], [0.9044196305, -0.3014910643, -0.3018745932, 5500.0], [0.9044195348, 0.3014912943, -0.3018746504, 5500.0], [0.3014912367, -0.9044193748, -0.3018751873, 5500.0], [0.3015110819, -0.3013465199, -0.9045890462, 5500.0], [0.3014815607, 0.3014810386, -0.9045540625, 5500.0], [0.301491691, 0.9044192968, -0.3018749673, 5500.0], [-0.3014917901, -0.9044192099, -0.3018751286, 5500.0], [-0.3015110626, -0.3013465021, -0.9045890585, 5500.0], [-0.3014814747, 0.3014810781, -0.904554078, 5500.0], [-0.301491225, 0.904419444, -0.3018749917, 5500.0], [-0.9044195788, -0.3014910964, -0.301874716, 5500.0], [-0.9044196065, 0.3014912379, -0.3018744917, 5500.0], [0.5772539059, -0.577254112, -0.5775427416, 6000.0], [0.5772537687, 0.5772545673, -0.5775424235, 6000.0], [-0.5772541227, -0.5772538622, -0.5775427744, 6000.0], [-0.577254148, 0.5772543081, -0.5775423035, 6000.0], [0.8320501259, -0.5547003438, -0.0003413330173, 6500.0], [0.8319060448, 3.098869511e-08, -0.5549165096, 6500.0], [0.5546999144, -0.8320504122, -0.0003413904583, 6500.0], [0.554572481, -4.940954412e-08, -0.8321354236, 6500.0], [6.072960914e-10, -0.8319059468, -0.5549166565, 6500.0], [-3.666727552e-10, -0.5545724001, -0.8321354776, 6500.0], [2.11767819e-10, 0.5545717321, -0.8321359228, 6500.0], [-3.600334341e-10, 0.831906107, -0.5549164164, 6500.0], [-0.5546999789, -0.8320503692, -0.0003413141108, 6500.0], [-0.5545722601, 2.913643411e-08, -0.8321355709, 6500.0], [-0.832050099, -0.5547003842, -0.0003413181217, 6500.0], [-0.8319060534, 3.281216423e-10, -0.5549164967, 6500.0], [0.8017208664, -0.53446813, -0.2675583495, 7000.0], [0.8016418421, -0.2672396717, -0.5347460283, 7000.0], [0.8016418981, 0.2672394531, -0.5347460537, 7000.0], [0.8017207264, 0.534468159, -0.267558711, 7000.0], [0.5344681068, -0.8017208029, -0.2675585861, 7000.0], [0.53442475, -0.2671552934, -0.8018841786, 7000.0], [0.5343916032, 0.2672533946, -0.8018735795, 7000.0], [0.5344678393, 0.8017210021, -0.2675585238, 7000.0], [0.2672396447, -0.8016419395, -0.5347458958, 7000.0], [0.2672530582, -0.5343914545, -0.8018737907, 7000.0], [0.2672532447, 0.5343918496, -0.8018734652, 7000.0], [0.2672394535, 0.8016416949, -0.534746358, 7000.0], [-0.2672393738, -0.8016420235, -0.5347459053, 7000.0], [-0.2672529745, -0.5343914835, -0.8018737993, 7000.0], [-0.2672527998, 0.534391969, -0.801873534, 7000.0], [-0.2672395729, 0.8016417046, -0.5347462838, 7000.0], [-0.534468046, -0.8017208702, -0.2675585059, 7000.0], [-0.534424298, -0.2671553634, -0.8018844565, 7000.0], [-0.5343916999, 0.2672534444, -0.8018734985, 7000.0], [-0.5344680464, 0.801720919, -0.267558359, 7000.0], [-0.8017210357, -0.5344679886, -0.2675581246, 7000.0], [-0.8016417412, -0.2672397248, -0.5347461531, 7000.0], [-0.8016420535, 0.2672394791, -0.5347458078, 7000.0], [-0.8017211137, 0.5344675679, -0.2675587314, 7000.0], [2.651765635e-12, -0.9999999687, -0.0002500086681, 8000.0], [1.455731894e-10, 2.590861083e-08, -1.0, 8000.0], [-0.9999999687, -1.852948745e-11, -0.0002500002363, 8000.0]], 'mrtrix_version': '3.0.2'}}
dwibiascorrect: [DEBUG] path.name_temporary() (from image.py:33): /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mrtrix-tmp-6azLV9.json
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): mrinfo -> mrinfo
dwibiascorrect: [DEBUG] run.version_match() (from image.py:34): Version-matched executable for mrinfo: /home/localadmin/Softwares/miniconda3/bin/mrinfo
dwibiascorrect: [DEBUG] run.exe_name() (from image.py:34): /home/localadmin/Softwares/miniconda3/bin/mrinfo -> /home/localadmin/Softwares/miniconda3/bin/mrinfo
dwibiascorrect: Loading header for image file 'mask.mif'
dwibiascorrect: [DEBUG] image.__init__() (from image.py:179): ['/home/localadmin/Softwares/miniconda3/bin/mrinfo', 'mask.mif', '-json_all', '/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mrtrix-tmp-6azLV9.json']
dwibiascorrect: [DEBUG] image.__init__() (from image.py:179): {'_name': 'mask.mif', '_size': [96, 96, 38], '_spacing': [2.20833, 2.20833, 3.0], '_strides': [1, 2, 3], '_format': 'MRtrix', '_datatype': 'Bit', '_intensity_offset': 0.0, '_intensity_scale': 1.0, '_transform': [[0.999999943880049, 5.33884719919637e-08, 0.000335022184972424, -102.900543212891], [1.57276657824971e-05, 0.998889975224889, -0.0471043187585057, -93.3131484985352], [-0.00033465286422834, 0.0471043245914531, 0.998889919317656, -23.3364963531494], [0.0, 0.0, 0.0, 1.0]], '_keyval': {'command_history': '/home/localadmin/Softwares/miniconda3/bin/mrconvert /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mask.mif -datatype bit -info  (version=3.0.2)', 'comments': 'FSL5.0', 'mrtrix_version': '3.0.2'}}
dwibiascorrect: [DEBUG] image.match() (from dwibiascorrect:69): 'in.mif' 'mask.mif' image match
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): dwiextract -> dwiextract
dwibiascorrect: [DEBUG] run.version_match() (from run.py:330): Version-matched executable for dwiextract: /home/localadmin/Softwares/miniconda3/bin/dwiextract
dwibiascorrect: [DEBUG] run._shebang() (from run.py:340): File "/home/localadmin/Softwares/miniconda3/bin/dwiextract": Not a text file
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): mrmath -> mrmath
dwibiascorrect: [DEBUG] run.version_match() (from run.py:330): Version-matched executable for mrmath: /home/localadmin/Softwares/miniconda3/bin/mrmath
dwibiascorrect: [DEBUG] run._shebang() (from run.py:340): File "/home/localadmin/Softwares/miniconda3/bin/mrmath": Not a text file
dwibiascorrect: [DEBUG] run.command() (from ants.py:64): To execute: [['/home/localadmin/Softwares/miniconda3/bin/dwiextract', 'in.mif', '-', '-bzero', '-info'], ['/home/localadmin/Softwares/miniconda3/bin/mrmath', '-', 'mean', 'mean_bzero.mif', '-axis', '3', '-info']]
Command:  dwiextract in.mif - -bzero | mrmath - mean mean_bzero.mif -axis 3
          dwiextract: [INFO] opening image "in.mif"...
          dwiextract: [INFO] image "in.mif" opened with dimensions 96x96x38x129, voxel spacing 2.2083300000000001x2.2083300000000001x3x5.9000000000000004, datatype Float32LE
          dwiextract: [INFO] found 129x4 diffusion gradient table
          dwiextract: [INFO] creating image "-"...
          dwiextract: [INFO] image "/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mrtrix-tmp-sK5XHm.mif" created with dimensions 96x96x38x1, voxel spacing 2.2083300000000001x2.2083300000000001x3x5.9000000000000004, datatype Float32LE
          dwiextract: [100%] extracting volumes
          mrmath: [INFO] opening image "-"...
          mrmath: [INFO] image "/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/mrtrix-tmp-sK5XHm.mif" opened with dimensions 96x96x38x1, voxel spacing 2.2083300000000001x2.2083300000000001x3x5.9000000000000004, datatype Float32LE
          mrmath: [INFO] creating image "mean_bzero.mif"...
          mrmath: [INFO] image "mean_bzero.mif" created with dimensions 96x96x38, voxel spacing 2.2083300000000001x2.2083300000000001x3, datatype Float32LE
          mrmath: [100%] computing mean along axis 3...
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): mrconvert -> mrconvert
dwibiascorrect: [DEBUG] run.version_match() (from run.py:330): Version-matched executable for mrconvert: /home/localadmin/Softwares/miniconda3/bin/mrconvert
dwibiascorrect: [DEBUG] run._shebang() (from run.py:340): File "/home/localadmin/Softwares/miniconda3/bin/mrconvert": Not a text file
dwibiascorrect: [DEBUG] run.command() (from ants.py:68): To execute: [['/home/localadmin/Softwares/miniconda3/bin/mrconvert', 'mean_bzero.mif', 'mean_bzero.nii', '-strides', '+1,+2,+3', '-info']]
Command:  mrconvert mean_bzero.mif mean_bzero.nii -strides +1,+2,+3
          mrconvert: [INFO] opening image "mean_bzero.mif"...
          mrconvert: [INFO] image "mean_bzero.mif" opened with dimensions 96x96x38, voxel spacing 2.2083300000000001x2.2083300000000001x3, datatype Float32LE
          mrconvert: [INFO] no valid diffusion gradient table found
          mrconvert: [INFO] error importing diffusion gradient table for image "mean_bzero.mif"
          mrconvert: [INFO] opening image "+1,+2,+3"...
          mrconvert: [INFO] creating image "mean_bzero.nii"...
          mrconvert: [INFO] image "mean_bzero.nii" created with dimensions 96x96x38, voxel spacing 2.2083300000000001x2.2083300000000001x3, datatype Float32LE
          mrconvert: [100%] copying from "mean_bzero.mif" to "mean_bzero.nii"
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): mrconvert -> mrconvert
dwibiascorrect: [DEBUG] run.version_match() (from run.py:330): Version-matched executable for mrconvert: /home/localadmin/Softwares/miniconda3/bin/mrconvert
dwibiascorrect: [DEBUG] run._shebang() (from run.py:340): File "/home/localadmin/Softwares/miniconda3/bin/mrconvert": Not a text file
dwibiascorrect: [DEBUG] run.command() (from ants.py:69): To execute: [['/home/localadmin/Softwares/miniconda3/bin/mrconvert', 'mask.mif', 'mask.nii', '-strides', '+1,+2,+3', '-info']]
Command:  mrconvert mask.mif mask.nii -strides +1,+2,+3
          mrconvert: [INFO] opening image "mask.mif"...
          mrconvert: [INFO] image "mask.mif" opened with dimensions 96x96x38, voxel spacing 2.2083300000000001x2.2083300000000001x3, datatype Bit
          mrconvert: [INFO] no valid diffusion gradient table found
          mrconvert: [INFO] error importing diffusion gradient table for image "mask.mif"
          mrconvert: [INFO] opening image "+1,+2,+3"...
          mrconvert: [INFO] creating image "mask.nii"...
          mrconvert: [WARNING] requested datatype (Bit) not supported - substituting with UInt8
          mrconvert: [INFO] image "mask.nii" created with dimensions 96x96x38, voxel spacing 2.2083300000000001x2.2083300000000001x3, datatype UInt8
          mrconvert: [100%] copying from "mask.mif" to "mask.nii"
dwibiascorrect: [DEBUG] run.exe_name() (from run.py:546): N4BiasFieldCorrection -> N4BiasFieldCorrection
dwibiascorrect: [DEBUG] run.version_match() (from run.py:330): Version-matched executable for N4BiasFieldCorrection: /home/localadmin/Softwares/miniconda3/bin/N4BiasFieldCorrection
dwibiascorrect: [DEBUG] run._shebang() (from run.py:340): File "/home/localadmin/Softwares/miniconda3/bin/N4BiasFieldCorrection": Not a text file
dwibiascorrect: [DEBUG] run.command() (from ants.py:72): To execute: [['/home/localadmin/Softwares/miniconda3/bin/N4BiasFieldCorrection', '-d', '3', '-i', 'mean_bzero.nii', '-w', 'mask.nii', '-o', '[corrected.nii,init_bias.nii]', '-s', '4', '-b', '[100,3]', '-c', '[1000,0.0]', '-info']]
Command:  N4BiasFieldCorrection -d 3 -i mean_bzero.nii -w mask.nii -o [corrected.nii,init_bias.nii] -s 4 -b [100,3] -c [1000,0.0]
          ERROR:  Invalid command line flags found! Aborting execution.

dwibiascorrect: [ERROR] N4BiasFieldCorrection -d 3 -i mean_bzero.nii -w mask.nii -o [corrected.nii,init_bias.nii] -s 4 -b [100,3] -c [1000,0.0] (ants.py:72)
dwibiascorrect: [ERROR] Information from failed command:
dwibiascorrect:
                ERROR:  Invalid flag provided in
                ERROR:  Invalid command line flags found! Aborting execution.
dwibiascorrect:
dwibiascorrect: [ERROR] For debugging, inspect contents of scratch directory: /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/
dwibiascorrect: Scratch directory retained; location: /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-XEGOZD/
```

Looking at the line:
```
dwibiascorrect: [DEBUG] run.command() (from ants.py:72): To execute: [['/home/localadmin/Softwares/miniconda3/bin/N4BiasFieldCorrection', '-d', '3', '-i', 'mean_bzero.nii', '-w', 'mask.nii', '-o', '[corrected.nii,init_bias.nii]', '-s', '4', '-b', '[100,3]', '-c', '[1000,0.0]', '-info']]
```
It seems an extra option flag `-info` is added to the N4BiasFieldCorrection.

Running `dwibiascorrect` without the flag `-debug` succeeds, see below.

Command:
```bash
/home/localadmin/Softwares/miniconda3/bin/dwibiascorrect ants -mask /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz -bias diffusion_denoised_biasfield.mif -force /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_biascorrect/diffusion_denoised_biascorr.mif
dwibiascorrect: 
dwibiascorrect: Note that this script makes use of commands / algorithms that have relevant articles for citation; INCLUDING FROM EXTERNAL SOFTWARE PACKAGES. Please consult the help page (-help option) for more information.
dwibiascorrect: 
dwibiascorrect: [WARNING] Output file 'diffusion_denoised_biasfield.mif' already exists; will be overwritten at script completion
dwibiascorrect: Generated scratch directory: /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-683WFE/
Command:  mrconvert /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_denoise/diffusion_denoised.mif /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-683WFE/in.mif
Command:  mrconvert /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/flirt_dwimask/dwi_brain_mask.nii.gz /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-683WFE/mask.mif -datatype bit
dwibiascorrect: Changing to scratch directory (/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-683WFE/)
Command:  dwiextract in.mif - -bzero | mrmath - mean mean_bzero.mif -axis 3
Command:  mrconvert mean_bzero.mif mean_bzero.nii -strides +1,+2,+3
Command:  mrconvert mask.mif mask.nii -strides +1,+2,+3
Command:  N4BiasFieldCorrection -d 3 -i mean_bzero.nii -w mask.nii -o [corrected.nii,init_bias.nii] -s 4 -b [100,3] -c [1000,0.0]
Command:  mrcalc mean_bzero.mif mask.mif -mult - | mrmath - sum - -axis 0 | mrmath - sum - -axis 1 | mrmath - sum - -axis 2 | mrdump -
Command:  mrcalc corrected.nii mask.mif -mult - | mrmath - sum - -axis 0 | mrmath - sum - -axis 1 | mrmath - sum - -axis 2 | mrdump -
Command:  mrcalc init_bias.nii 0.9515317417482443 -mult bias.mif
Command:  mrcalc in.mif bias.mif -div result.mif
Command:  mrconvert result.mif /home/localadmin/Desktop/HagmannTest/BIDS_out/ds-DSI/derivatives/nipype/sub-01/ses-01/diffusion_pipeline/preprocessing_stage/dwi_biascorrect/diffusion_denoised_biascorr.mif
Command:  mrconvert bias.mif /home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/diffusion_denoised_biasfield.mif
dwibiascorrect: Changing back to original directory (/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88)
dwibiascorrect: Deleting scratch directory (/home/localadmin/.local/share/Trash/files/dwibiascorrect-tmp-I6CQ88/dwibiascorrect-tmp-683WFE/)

```


**Platform/Environment/Version**

Please provide the following information:

-  OS: 
   Distributor ID:	Ubuntu
   Description:	Ubuntu 16.04.6 LTS
   Release:	16.04
   Codename:	xenial


-  *MRtrix3* version:
   == mrinfo 3.0.2 ==
   64 bit release version, built Sep 30 2020, using Eigen 3.3.7
   Author(s): J-Donald Tournier (d.tournier@brain.org.au) and Robert E. Smith (robert.smith@florey.edu.au)
  Copyright (c) 2008-2020 the MRtrix3 contributors.


---

--- a/core/app.cpp
+++ b/core/app.cpp
@@ -93,10 +93,20 @@ namespace MR
     std::string command_history_string;
     vector<ParsedArgument> argument;
     vector<ParsedOption> option;
+
     //ENVVAR name: MRTRIX_QUIET
     //ENVVAR Do not display information messages or progress status. This has
-    //ENVVAR the same effect as the ``-quiet`` command-line option.
-    int log_level = getenv("MRTRIX_QUIET") ? 0 : 1;
+    //ENVVAR the same effect as the ``-quiet`` command-line option. If set,
+    //ENVVAR supersedes the MRTRIX_LOGLEVEL environment variable.
+
+    //ENVVAR name: MRTRIX_LOGLEVEL
+    //ENVVAR Set the default terminal verbosity. Default terminal verbosity
+    //ENVVAR is 1. This has the same effect as the ``-quiet`` (0),
+    //ENVVAR ``-info`` (2) or ``-debug`` (3) comand-line options.
+    int log_level = getenv("MRTRIX_QUIET") ?
+                    0 :
+                    (getenv("MRTRIX_LOGLEVEL") ? to<int>(getenv("MRTRIX_LOGLEVEL")) : 1);
+
     int exit_error_code = 0;
     bool fail_on_warn = false;
     bool terminal_use_colour = true;

--- a/docs/reference/environment_variables.rst
+++ b/docs/reference/environment_variables.rst
@@ -32,6 +32,12 @@ List of MRtrix3 environment variables
      ``/etc`` folder is problematic, or to allow different versions of
      the software to have different configurations, etc.
 
+.. envvar:: MRTRIX_LOGLEVEL
+
+     Set the default terminal verbosity. Default terminal verbosity
+     is 1. This has the same effect as the ``-quiet`` (0),
+     ``-info`` (2) or ``-debug`` (3) comand-line options.
+
 .. envvar:: MRTRIX_NOSIGNALS
 
      If this variable is set to any value, disable MRtrix3's custom
@@ -50,7 +56,8 @@ List of MRtrix3 environment variables
 .. envvar:: MRTRIX_QUIET
 
      Do not display information messages or progress status. This has
-     the same effect as the ``-quiet`` command-line option.
+     the same effect as the ``-quiet`` command-line option. If set,
+     supersedes the MRTRIX_LOGLEVEL environment variable.
 
 .. envvar:: MRTRIX_RNG_SEED
 

--- a/lib/mrtrix3/app.py
+++ b/lib/mrtrix3/app.py
@@ -14,7 +14,6 @@
 # For more details, see http://www.mrtrix.org/.
 
 import argparse, inspect, math, os, random, shlex, shutil, signal, string, subprocess, sys, textwrap, time
-import mrtrix3
 from mrtrix3 import ANSI, CONFIG, MRtrixError, setup_ansi
 from mrtrix3 import utils # Needed at global level
 from ._version import __version__

--- a/lib/mrtrix3/app.py
+++ b/lib/mrtrix3/app.py
@@ -43,7 +43,7 @@ EXEC_NAME = os.path.basename(sys.argv[0])
 FORCE_OVERWRITE = False #pylint: disable=unused-variable
 NUM_THREADS = None #pylint: disable=unused-variable
 SCRATCH_DIR = ''
-VERBOSITY = 0 if 'MRTRIX_QUIET' in mrtrix3.CONFIG else 1
+VERBOSITY = 0 if 'MRTRIX_QUIET' in os.environ else int(os.environ.get('MRTRIX_LOGLEVEL', '1'))
 WORKING_DIR = os.getcwd()
 
 
@@ -187,7 +187,7 @@ def _execute(module): #pylint: disable=unused-variable
       pass
     run.shared.set_continue(ARGS.cont[1])
 
-  run.shared.verbosity = VERBOSITY
+  run.shared.set_verbosity(VERBOSITY)
   run.shared.set_num_threads(NUM_THREADS)
 
   CMDLINE.print_citation_warning()
@@ -486,6 +486,8 @@ class ProgressBar(object): #pylint: disable=unused-variable
     self.wrapoff = '' if self.newline else ProgressBar.WRAPOFF
     self.wrapon = '' if self.newline else ProgressBar.WRAPON
     VERBOSITY = run.shared.verbosity = VERBOSITY - 1 if VERBOSITY else 0
+    if not self.orig_verbosity:
+      return
     if self.isatty:
       sys.stderr.write(self.wrapoff + EXEC_NAME + ': ' + ANSI.execute + '[' + ('{0:>3}%'.format(self.value) if self.multiplier else ProgressBar.BUSY[0]) + ']' + ANSI.clear + ' ' + ANSI.console + self._get_message() + '... ' + ANSI.clear + ANSI.lineclear + self.wrapon + self.newline)
     else:
@@ -523,6 +525,9 @@ class ProgressBar(object): #pylint: disable=unused-variable
       self.message = msg
     if self.multiplier:
       self.value = 100
+    VERBOSITY = run.shared.verbosity = self.orig_verbosity
+    if not self.orig_verbosity:
+      return
     if self.isatty:
       sys.stderr.write('\r' + EXEC_NAME + ': ' + ANSI.execute + '[' + ('100%' if self.multiplier else 'done') + ']' + ANSI.clear + ' ' + ANSI.console + self._get_message() + ANSI.clear + ANSI.lineclear + '\n')
     else:
@@ -531,11 +536,13 @@ class ProgressBar(object): #pylint: disable=unused-variable
       else:
         sys.stderr.write('=' * (int(self.value/2) - int(self.old_value/2)) + ']\n')
     sys.stderr.flush()
-    VERBOSITY = run.shared.verbosity = self.orig_verbosity
+
 
   def _update(self):
     global EXEC_NAME
     assert not self.iscomplete
+    if not self.orig_verbosity:
+      return
     if self.isatty:
       sys.stderr.write(self.wrapoff + '\r' + EXEC_NAME + ': ' + ANSI.execute + '[' + ('{0:>3}%'.format(self.value) if self.multiplier else ProgressBar.BUSY[self.counter%6]) + ']' + ANSI.clear + ' ' + ANSI.console + self._get_message() + '... ' + ANSI.clear + ANSI.lineclear + self.wrapon + self.newline)
     else:

--- a/lib/mrtrix3/image.py
+++ b/lib/mrtrix3/image.py
@@ -240,7 +240,12 @@ def statistics(image_path, **kwargs): #pylint: disable=unused-variable
   if app.VERBOSITY > 1:
     app.console('Command: \'' + ' '.join(command) + '\' (piping data to local storage)')
 
-  proc = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=None)
+  try:
+    from subprocess import DEVNULL
+  except ImportError:
+    import os
+    DEVNULL = open(os.devnull, 'wb')
+  proc = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=DEVNULL)
   stdout = proc.communicate()[0]
   if proc.returncode:
     raise MRtrixError('Error trying to calculate statistics from image \'' + image_path + '\'')

--- a/lib/mrtrix3/image.py
+++ b/lib/mrtrix3/image.py
@@ -241,9 +241,8 @@ def statistics(image_path, **kwargs): #pylint: disable=unused-variable
     app.console('Command: \'' + ' '.join(command) + '\' (piping data to local storage)')
 
   try:
-    from subprocess import DEVNULL
+    from subprocess import DEVNULL #pylint: disable=import-outside-toplevel
   except ImportError:
-    import os
     DEVNULL = open(os.devnull, 'wb')
   proc = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=DEVNULL)
   stdout = proc.communicate()[0]

--- a/lib/mrtrix3/run.py
+++ b/lib/mrtrix3/run.py
@@ -55,6 +55,14 @@ class Shared(object):
     if self.env.get('SGE_ROOT') and self.env.get('JOB_ID'):
       del self.env['SGE_ROOT']
 
+    # If environment variable is set, should apply to invoked script,
+    #   but not to any underlying invoked commands
+    try:
+      self.env.pop('MRTRIX_QUIET')
+    except KeyError:
+      pass
+    self.env['MRTRIX_LOGLEVEL'] = 1
+
     # Flagged by calling the set_continue() function;
     #   run.command() and run.function() calls will be skipped until one of the inputs to
     #   these functions matches the given string
@@ -71,7 +79,7 @@ class Shared(object):
     self.process_lists = [ ]
 
     self._scratch_dir = None
-    self.verbosity = 0
+    self.verbosity = 1
 
   # Acquire a unique index
   # This ensures that if command() is executed in parallel using different threads, they will
@@ -144,6 +152,14 @@ class Shared(object):
   def set_scratch_dir(self, path):
     self.env['MRTRIX_TMPFILE_DIR'] = path
     self._scratch_dir = path
+
+  # Controls verbosity of invoked MRtrix3 commands, as well as whether or not the
+  #   stderr outputs of invoked commands are propagated to the terminal instantly or
+  #   instead written to a temporary file for read on completion
+  def set_verbosity(self, verbosity):
+    assert isinstance(verbosity, int)
+    self.verbosity = verbosity
+    self.env['MRTRIX_LOGLEVEL'] = str(max(1, verbosity-1))
 
   # Terminate any and all running processes, and delete any associated temporary files
   def terminate(self, signum): #pylint: disable=unused-variable
@@ -330,9 +346,6 @@ def command(cmd, **kwargs): #pylint: disable=unused-variable
         line[0] = version_match(line[0])
         if shared.get_num_threads() is not None:
           line.extend( [ '-nthreads', str(shared.get_num_threads()) ] )
-        # Get MRtrix3 binaries to output additional INFO-level information if script running in debug mode
-        if shared.verbosity == 3 and not any(entry in line for entry in ['-info', '-debug']):
-          line.append('-info')
         if force:
           line.append('-force')
       else:

